packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -339,25 +339,30 @@ class AppDomain extends Domain {
       );
     }
 
+    bool supportsRestart = hotMode ? device.supportsHotMode : device.supportsRestart;
```

Comment:
Hot mode makes the distinction between reload and restart and that really matters for the user experience. Can this distinction be made here?